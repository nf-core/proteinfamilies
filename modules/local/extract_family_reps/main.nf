process EXTRACT_FAMILY_REPS {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/eb/eb3700531c7ec639f59f084ab64c05e881d654dcf829db163539f2f0b095e09d/data' :
        'community.wave.seqera.io/library/biopython:1.84--3318633dad0031e7' }"

    input:
    tuple val(meta), path(aln, stageAs: "aln/*")

    output:
    tuple val(meta), path("${meta.id}_reps.fa")     , emit: fasta
    tuple val(meta), path("${meta.id}_meta_mqc.csv"), emit: map
    path "versions.yml"                             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    extract_family_reps.py \\
        --full_msa_folder aln \\
        --metadata ${prefix}_meta_mqc.csv \\
        --out_fasta ${prefix}_reps.fa

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
        biopython: \$(python -c "import importlib.metadata; print(importlib.metadata.version('biopython'))")
    END_VERSIONS
    """
}
