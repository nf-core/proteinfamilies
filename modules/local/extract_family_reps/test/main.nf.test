nextflow_process {

    name "Test Process EXTRACT_FAMILY_REPS"
    script "../main.nf"
    process "EXTRACT_FAMILY_REPS"

    test("proteinfamilies - faa") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'extract_family_reps' ],
                    [
                        file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_1.faa', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_2.faa', checkIfExists: true)
                    ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.fasta[0][1]).name,
                    path(process.out.fasta[0][1]).readLines().size(),
                    file(process.out.map[0][1]).name,
                    path(process.out.map[0][1]).readLines().size(),
                    process.out.versions.collect { path(it).yaml }
                    ).match()}
            )
        }
    }

    test("proteinfamilies - faa - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'extract_family_reps' ],
                    [
                        file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_1.faa', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_2.faa', checkIfExists: true)
                    ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
