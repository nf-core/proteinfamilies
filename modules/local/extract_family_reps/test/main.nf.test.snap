{
    "proteinfamilies - faa": {
        "content": [
            "extract_family_reps_reps.faa",
            4,
            "extract_family_reps_meta_mqc.csv",
            8,
            [
                {
                    "EXTRACT_FAMILY_REPS": {
                        "python": "3.13.1"
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-29T11:25:17.775635822"
    },
    "proteinfamilies - faa - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "extract_family_reps"
                        },
                        "extract_family_reps_reps.faa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "extract_family_reps"
                        },
                        "extract_family_reps_meta_mqc.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,3f12f02c71ad2d3d5aff7885537cbd9f"
                ],
                "fasta": [
                    [
                        {
                            "id": "extract_family_reps"
                        },
                        "extract_family_reps_reps.faa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "map": [
                    [
                        {
                            "id": "extract_family_reps"
                        },
                        "extract_family_reps_meta_mqc.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3f12f02c71ad2d3d5aff7885537cbd9f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-29T11:25:31.692612332"
    }
}